.github/workflows/compile-platform-examples.yml

```diff
@@ -149,7 +158,7 @@ jobs:
           platforms: |
             # Use Boards Manager to install the latest release of the platform to get the toolchain.
             - name: Infineon:xmc
-              source-url: https://github.com/Infineon/XMC-for-Arduino/releases/latest/download/package_infineon_index.json
+              source-url: https://github.com/LinjingZhang/XMC-for-Arduino/releases/download/2.2.1/package_infineon_index_linjing_test.json

```

Comment:
This might not work because you are renaming the .json file. 
As the preference configuration is meant to use "package_infineon_index.json" your file would not be find any board. You can give it a try to change that file name in your releases and update then correrspondingly the yml file. 
Basically, only the repo owner in the url should be changed : "Infineon" for "LinjingZhang" if you upload the modified json to your latest release in your fork.

Reply:
Workflow error solved:
1. Change index.json file name
2. Manual adapt files for release
3. For arduino/compile-sketches@v1, disable 'enable-deltas-report'
4. Some test is not compatible (e.g. stackMem, linker script changed)